traitsui/qt4/tabular_editor.py

```diff
@@ -929,8 +929,7 @@ def columnResized(self, index, old, new):
             if self._user_widths is None:
                 self._user_widths = [None] * len(self._editor.adapter.columns)
             self._user_widths[index] = new
-            if (self._editor.factory is not None
```

Comment:
I _could_ keep this but I'd prefer not. Other places in this class uses `factory` without checking if the value is None.

Reply:
Aha this causes the integration tests (which open and close a lot of GUI) to have this failure in a subtest that has nothing to do with tabular editor:
```
FAIL: test_run (test_all_examples.TestExample) (file_path='/Users/appveyor/.edm/envs/traitsui-test-3.6-pyqt/lib/python3.6/site-packages/traitsui/examples/demo/Advanced/Time_editor_demo.py')
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/appveyor/projects/traitsui/integrationtests/test_all_examples.py", line 295, in test_run
    run_file(file_path)
RuntimeError: Uncaught exceptions found.
=== Exception (type: <class 'AttributeError'>, value: 'NoneType' object has no attribute 'auto_resize') ===
Traceback (most recent call last):
  File "/Users/appveyor/.edm/envs/traitsui-test-3.6-pyqt/lib/python3.6/site-packages/traitsui/qt4/tabular_editor.py", line 932, in columnResized
    if not self._editor.factory.auto_resize:
AttributeError: 'NoneType' object has no attribute 'auto_resize'
```
This is in Mac OSX and Qt4. Note that the `Time_editor_demo` has nothing to do with TabularEditor. This is clearly caused by events being left unprocessed at the end of a test.  We have already identified the approach to process cascaded events not to work in OSX and Qt4 (see #951) despite Qt says it should (really, Qt5 says it should).

Reply:
Confirmed this is Qt4 refusing to process all cascaded events in tests. The tests are behaving fine with Qt5 and other platforms. I will put this back for now as there are downstream projects still testing with Qt4. This is the wrong place to do this sort of thing but it is cheap.